Modules/posixmodule.c

```diff
@@ -5490,12 +5490,102 @@ os__path_normpath_impl(PyObject *module, PyObject *path)
         return NULL;
     }
     Py_ssize_t norm_len;
-    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, &norm_len);
+    wchar_t *norm_path = _Py_normpath_and_size(buffer, len, 0, &norm_len);
     PyObject *result = PyUnicode_FromWideChar(norm_path, norm_len);
     PyMem_Free(buffer);
     return result;
 }
 
+#ifndef MS_WINDOWS
+
+/*[clinic input]
+os._path_abspath
+
+    path: unicode
+    /
+
+Make path absolute.
+[clinic start generated code]*/
+
+static PyObject *
+os__path_abspath_impl(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=b58956d662b60be0 input=577ecb3473d22113]*/
+{
+    Py_ssize_t path_len, start, abs_len, cwd_len;
+    wchar_t *abs, *abs_buf = NULL, *cwd_buf = NULL;
+    PyObject *result = NULL;
```

Comment:
Looking at this list of variables, I think we could benefit of making some of them more narrowly scoped. For example, `cwd_len` seems to be used only in the `else` body in L5534-L5569, so I'd consider moving that declaration into the `else` body. Ditto for `cwd_buf` (and its related error handling).

Reply:
Either we need to free `cwd_buf` for all exits in the `else` body, or we have to put in back in the function scope :)

Reply:
Thanks. I don't see a clean way to do it without the broad scope.